src/mumble/ServerHandler.cpp

```diff
@@ -392,6 +393,11 @@ void ServerHandler::sendPing() {
 	if (!connection)
 		return;
 
+	if (iInFlightTCPPings >= g.s.iMaxInFlightTCPPings) {
+		serverConnectionClosed(QAbstractSocket::UnknownSocketError, tr("Server is not responding to TCP pings"));
```

Comment:
So we don’t close the `QSslSocket` via `connection->disconnect(true)` because we expect it to be dead already/nothing to clean up?
If we as a client are not aware of the dropped TCP connection, shouldn’t we `abort` the `QSslSocket`?

That would also emit the closed signal - in a cleaner way (responsibility of the connection to emit the closed signal).